woocommerce/woocommerce-ios
# Networking & Yosemite layers: add `buttonText` to `Product`

Networking & Yosemite layers for #2200 

- [x] ⚠️ depends on schema update in the storage layer https://github.com/woocommerce/woocommerce-ios/pull/2391

## Changes

- Added `buttonText: String` to the readonly `Product` in the Networking layer
  - Updated a bunch of usage in unit tests in `ProductUpdater` (wouldn't have to do this with `Copiable`!)
- Networking layer:
  - Added `product-external.json` as a sample API response from retrieving an external product
  - Added a test case for loading a product
- Yosemite layer:
  - Based on our discussion in https://github.com/woocommerce/woocommerce-ios/pull/2387/files#r435272569, I experimented with `ProductsEndpointsProviding`/`MockProductsRemote` (especially since they were already created 😄). Added an initializer with `remote: ProductsEndpointsProviding` parameter with `ProductsRemote(network: network)` as default, and used the `remote` for loading a product for a new test case in `ProductStoreTests`

## Testing

Just a sanity check on fetching products:

- Launch the app
- Go to the Orders tab
- Tap on an order with a simple product
- Tap on a simple product --> the product details should launch as before

Update release notes:
